addons/website/static/src/js/tours/tour_utils.js

```diff
@@ -471,6 +471,38 @@ function switchWebsite(websiteId, websiteName) {
     }];
 }
 
+/**
+ * Toggles the mobile preview.
+ *
+ * @returns {Array}
+ */
+function toggleMobilePreview() {
+    return [{
+        content: "Toggle the mobile preview",
+        trigger: ".o_we_website_top_actions [data-action='mobile']",
+    }, {
+        content: "Check that the mobile preview is toggled",
+        trigger: "iframe #wrapwrap.o_is_mobile",
+        isCheck: true,
+    }];
+}
+
+/**
+ * Toggles the desktop view.
+ *
+ * @returns {Array}
+ */
+function toggleDesktopView() {
```

Comment:
Good idea to put that in the utils. Both could be a single function
```js
function togglePreview(screenType) {
    return [{
        content: `Toggle the ${screenType} preview`,
        trigger: ".o_we_website_top_actions [data-action='mobile']",
    }, {
        content: `Check that the ${screenType} preview is toggled`,
        trigger: `iframe #wrapwrap${screenType === "mobile" ? ".o_is_mobile" : ":not(.o_is_mobile)"}`,
        isCheck: true,
    }];
}
```
(Incidentally, your functions could just be plain arrays.)

Reply:
I thought about making it a single function with a parameter, but after discussing with BSO, we found that it was easier to use if they were separated :slightly_smiling_face: 
Like in a tour, you directly understand what "toggleMobilePreview" and "toggleDesktopView" do, while "togglePreview" is not that straightforward (like the preview of what ?) and also nitpick but technically, the desktop view is not a preview so the name of the function would be horrible haha (like "toggleDesktopViewOrMobilePreview")
It could also be simply "toggleMobilePreview" with a boolean as parameter (like on/off, since it is the point of a toggle)
Anyway, it's a matter of taste I guess :shrug: 

> your functions could just be plain arrays

Not sure if you are asking to change it but a lot of other functions do like this too, so it is consistent with them :+1:

Reply:
But isn't `togglePreview("mobile")` / `togglePreview("desktop")` just as clear and readable? It looks a bit strange to me to have 2 separate functions/variables for that with such a similar code.

> technically, the desktop view is not a preview

I didn't even see the name difference. Actually I think that makes it worse, if there are 2 functions, I would expect them to only differ on the type. And I don't agree, the iframe _is_ a preview (I mean the files are called `website_preview`), so the desktop view is a preview.
But a single function could be named `toggleView()`, or `toggleIframeView()` because a preview is a view anyway.

> a lot of other functions do like this too, so it is consistent with them

From what I've seen (I may be wrong), the reason other functions do this is because it's needed in a lot of cases (what I have in mind is if you use `_t()` inside the array, or if a nested function in the array uses `_t()`, then you need to use a function). If it's not needed, I don't see why we should do it :thinking:

Reply:
I prefer parameters rather than code duplication in general too... but both are fine of course. If I have to decide I would actually go with `toggleMobile` (or `toggleMobilePreview` if you prefer) with a boolean as parameter. It's clear what it is about (more than `togglePreview`) and giving a boolean is more robust and "efficient" than using 2 strings.

If later we have more devices, I would go with `toggleDevice` and the parameter would be a defined constant in tour utils (`MOBILE`, `DESKTOP`, ...)... but no need to anticipate that IMO: at the moment "toggle + boolean" is the standard javascript way.

About the steps themselves:
```javascript
return [{
        content: "Toggle the desktop view",
        trigger: ".o_we_website_top_actions [data-action='mobile']",
    }, {
        content: "Check that the mobile preview is not toggled",
        trigger: "iframe #wrapwrap:not(.o_is_mobile)",
        isCheck: true,
    }];
```
=> Make sure to test `#wrapwrap.o_is_mobile` as an extra trigger of the first step, otherwise the second step is not robust (if it would be `#wrapwrap:not(typo)`, it would test the same thing).